internal/registry/registry_test.go

```diff
@@ -62,6 +71,131 @@ var _ = Describe("ImageExists", func() {
 			Expect(err.Error()).To(ContainSubstring("cannot get keychain from the registry auth getter"))
 		})
 	})
+
+	Context("Cannot get image manifest ", func() {
+
+		var err error
+
+		AfterEach(func() {
+			Expect(err.Error()).To(ContainSubstring("could not get image"))
+			Expect(err.Error()).To(ContainSubstring("failed to get manifest stream from image"))
+		})
+
+		It("should fail if it cannot get crane manifest", func() {
+
+			server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+				w.WriteHeader(http.StatusBadRequest)
+			}))
+			defer server.Close()
+			var u *url.URL
+			u, err = url.Parse(server.URL)
+			Expect(err).ToNot(HaveOccurred())
+			serverHost := strings.Split(u.String(), "/")[2]
+
+			endpoint := strings.Join(strings.Split(validImage, "/")[1:], "/")
+			image := fmt.Sprintf("%s/%s", serverHost, endpoint)
+			_, err = reg.ImageExists(ctx, image, ootov1alpha1.PullOptions{}, nil)
+
+			Expect(err).To(HaveOccurred())
+			Expect(err.Error()).To(ContainSubstring("failed to get crane manifest from image"))
+		})
+
+		It("should fail if it cannot unmarshal crane manifest", func() {
+
+			server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+				// return no error but with an empty manifest (which is not a valid manifest file)
+			}))
+			defer server.Close()
+			var u *url.URL
+			u, err = url.Parse(server.URL)
+			Expect(err).ToNot(HaveOccurred())
+			serverHost := strings.Split(u.String(), "/")[2]
+
+			endpoint := strings.Join(strings.Split(validImage, "/")[1:], "/")
+			image := fmt.Sprintf("%s/%s", serverHost, endpoint)
```

Comment:
```suggestion

			endpoint := strings.Join(strings.Split(validImage, "/")[1:], "/")
			image := fmt.Sprintf("%s/%s", u.Host, endpoint)
```
This applies to most tests in this PR.

Reply:
Done.